components/script/dom/svgsvgelement.rs

```diff
@@ -5,34 +5,103 @@
 use crate::dom::attr::Attr;
 use crate::dom::bindings::codegen::Bindings::SVGSVGElementBinding;
 use crate::dom::bindings::inheritance::Castable;
-use crate::dom::bindings::root::{DomRoot, LayoutDom};
+use crate::dom::bindings::reflector::DomObject;
+use crate::dom::bindings::root::{DomRoot, LayoutDom, Dom};
 use crate::dom::bindings::str::DOMString;
 use crate::dom::document::Document;
 use crate::dom::element::{AttributeMutation, Element, RawLayoutElementHelpers};
-use crate::dom::node::Node;
+use crate::dom::bindings::codegen::Bindings::WebGLRenderingContextBinding::WebGLRenderingContextConstants as constants;
+use crate::dom::node::{Node, window_from_node};
 use crate::dom::svggraphicselement::SVGGraphicsElement;
 use crate::dom::virtualmethods::VirtualMethods;
 use dom_struct::dom_struct;
 use html5ever::{LocalName, Prefix};
 use script_layout_interface::SVGSVGData;
 use style::attr::AttrValue;
+use canvas_traits::webgl::{WebGLMsg, WebGLContextShareMode, WebGLMsgSender, webgl_channel, WebGLVersion, GLContextAttributes, WebGLCommand};
+use std::cell::{Cell, RefCell, Ref};
+use euclid::Size2D;
+use crate::dom::window::Window;
+use crate::dom::bindings::cell::DomRefCell;
+use std::borrow::{Borrow, BorrowMut};
+use crate::dom::webglcontextevent::WebGLContextEvent;
+use crate::dom::event::{EventBubbles, EventCancelable, Event};
+use crate::dom::webglrenderingcontext::capture_webgl_backtrace;
+use crate::dom::bindings::codegen::Bindings::ElementBinding::ElementBinding::ElementMethods;
+use crate::dom::bindings::error::Fallible;
+use crate::dom::node::ChildrenMutation;
 
 const DEFAULT_WIDTH: u32 = 300;
 const DEFAULT_HEIGHT: u32 = 150;
 
 #[dom_struct]
 pub struct SVGSVGElement {
     svggraphicselement: SVGGraphicsElement,
+    #[ignore_malloc_size_of = "Channels are hard"]
+    webgl_sender: Option<WebGLMsgSender>,
+    share_mode: WebGLContextShareMode,
+    #[ignore_malloc_size_of = "Just a string"]
+    htmlString: Cell<Option<String>>,
```

Comment:
@jdm 

What's the best way to go about storing a String inside of a struct here? I'm getting the following compile error `error[E0599]: no method named `trace` found for type `&std::cell::Cell<std::option::Option<std::string::String>>` in the current scope
` for the current approach I'm taking.

Reply:
Use DomRefCell instead of Cell.

Reply:
Thanks!